### PR TITLE
[WIP] New auth

### PR DESCRIPTION
**Pull request by elitan**

### CODE REVIEW

File: src/shared/schema.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ const extendedJoi: ExtendedJoi = Joi.extend((joi) => ({
   }
 }))
 
-const passwordRule = Joi.string().min(6).max(128).required()
```
Comment (pnfcre) [inline src/shared/schema.ts]: I think the password rule should remain at minimum 6 characters, for security™ 😅

Comment (elitan) [reply src/shared/schema.ts]: Yes, make sense to default to 6 chars. But should maybe be an env var setting.

Comment (elitan) [reply src/shared/schema.ts]: https://github.com/nhost/hasura-backend-plus/issues/132

**Comment by plmercereau** [comment]
- renamed a lot from user to account, including the routes
- but then, which x-hasura-user-id are we using? user_id or account_id?
- account-user is defined as a ManyToOne. I think it should be OneToOne
- some cascade delete may need to be reviewed e.g. what happens when we delete a user (delete account, delete tokens) and when we delete an account (delete user, delete tokens)

**Review by hampuskraft — APPROVED** [review]

**Comment by elitan** [comment]
- I would say we use `public.users.id` as our primary `user_id` field.
- `public.users` <--> `auth.accounts` should be one to one yes. But isn't it that already? `auth.accunts.user_id` got a FK to `public.users.id`.
- Right, we might want to either not allow to delete an `auth.accounts` that got a `public.users` created. Or set a trigger to delete `public.users` when you delete an account on `auth.accounts`. Something like this: https://stackoverflow.com/a/22473228
- Right. I don't think we need to track that relationship really. The FK constraint is just to make sure that the value in `default_role` is present in the `account_roles` table.

Action points:

**How to handle DELETE for `auth.accounts`? Should we cascade deletes just like we do when we delete `public.users`?**

One way would be to set the FK the oppisite way from `accounts` --> `users`. And make the `accounts` table the parent. This makes sense since HBP could work without the `users` table, but not without the `accounts` table.

**Comment by elitan** [comment]
Note: We should probably set `auth.accounts.user_id` to `UNIQUE` to ensure that there is a 0..1 relations between `auth.accounts` and `public.users`.

This might change depending on what we decide given the discussions above.

**Pull request merged**